routes/drafts.ts

```diff
@@ -0,0 +1,79 @@
+import express from 'express';
+import DraftsService from '../services/drafts';
+import { SchemaValidationError } from 'slonik';
+import { formatQueryErrorResponse } from '../helpers';
+import { PendingDraft, DraftUpdate } from '../types/draft';
+
+const router = express.Router();
+
+router.get("/", async (req, res) => {
+  try{
+    const drafts = await DraftsService.index();
+    return res.status(200).json(drafts);
+  }catch(e){
+    if (e instanceof SchemaValidationError) {
+      return res.status(400)
+        .json({message: 'Error fetching drafts: ' + formatQueryErrorResponse(e)})
+    }
+
+    console.log(e)
+    return res.status(500).json({message: 'Server Error'})
+  }
+});
+
+router.post("/", async (req, res) => {
+  const data = req.body;
+  const validationResult = PendingDraft.safeParse(data);
+  if(!validationResult.success){
+    return res.status(422).json({message: 'Invalid data', error: validationResult.error})
+  }
+
+  try{
+    const draft = await DraftsService.store(data);
+    return res.status(201).json({ success: true, draft: JSON.stringify(draft)});
+  }catch(e){
+    console.log(e)
+    return res.status(500).json({message: 'Server Error'})
+  }
+});
+
+router.get("/:id", async (req, res) => {
+  let { id } = req.params;
+  try{
+    const draft = await DraftsService.show(id);
+    return res.status(200).json(draft);
+  }catch(e){
+    console.log(e)
+    return res.status(500).json({message: 'Server Error'})
+  }
+});
+
+router.put("/:id", async (req, res) => {
+  const { id } = req.params;
+  const data = req.body;
+  const validationResult = DraftUpdate.safeParse(data);
```

Comment:
Lets change this to a PendingDraft.safeParse now that we'll expect PUT to be a complete record and we can use DraftUpdate for the PATCH route. Both should be able to call the same update method on the server since we aren't going to handle upserts here.